### ISSUE TITLE
Capturing mouse click events

### ISSUE THREAD

**Issue by sergeyfridrikh** [bloomberg/bqplot#653]
After spending hours reading documentation and looking into examples of bqplot use cases I still can not answer a simple question: how to capture mouse position from a mouse click. Please help.

Also, if the bqplot team believes that the quote from documentation presented below (and there are plenty of those) helps users in any way they are missing something very important here.

Quote:

class bqplot.scales.OrdinalScale(**kwargs)
An ordinal scale.
A mapping from a discrete set of values to a numerical range.

**Comment by DougRzz** [comment]
Hi,
If any user believes that the documentation is not clear, constructive suggestions/contributions will be welcomed.  :-)
I'm not sure about the mouse position. I don't think its been implimented.  You are free to submit an issue.

**Comment by DougRzz** [comment]
I'm a mechanical engineer and not a software developer. So i understand what you are saying. I learnt bqplot last year using the examples in https://github.com/bloomberg/bqplot/tree/master/examples. The examples are really good. I believe there is a section where it describes how to change the size of the markers.  If you want to get creative with interactive plots (and you don't want to write JavaScript), bqplot is really good and it is worth learning. Hopefully the documentation will improve over time. Thanks for the feedback.

**Comment by andymcarter** [comment]
My 2c from a engineer turned recent software developer:

A combination of the tutorial notebooks and the docs (http://bqplot.readthedocs.io/) was how I learnt, first copy and pasting the examples from notebooks and gradually requiring the greater detail in the docs, understanding which kwargs belong to Axes, Scales, Marks etc. Learning the library from the docs alone would be challenging for me.

How did you end up setting the marker size? The parameter name 'default_size' might be the cause of some confusion, as it is 'default' for the plot you are making, not a bqplot-wide default that requires setting.

Searching the docs for 'scatter marker size' leads straight to the list of kwargs for the Scatter mark, and 'default_size' is listed. Explaining how to implement this in an API call feels a little out of scope. I originally thought the docs to be a bit parsimonious, but in fact they explain the API concisely, and leave any implementation to the notebooks.

**Comment by sergeyfridrikh** [comment]
@andymcarter  Exactly, I also use 'default_size' to set the marker size. Should it be this way? The name default_size becomes a misnomer if you can change the variable behind it and breeds confusion. Besides, why not give an example of how such a basic thing as setting marker size can be accomplished? 

Thanks for pointing me to the examples. They have been helpful, but definitely they are far from covering majority of typical use cases.

**Comment by DougRzz** [comment]
@sergeyfridrikh 
To set size in the scatter plot, see example below which is taken from the scatter.ipynb in the examples/marks folder.  Search for the sec1_returns variable which is a numpy array of sizes for each scatter point....

![image](https://user-images.githubusercontent.com/3510318/40549960-bb546dee-6030-11e8-8568-ccbf9acd41b1.png)


```python
from __future__ import print_function
import numpy as np
import pandas as pd
from bqplot import (
    Axis, ColorAxis, LinearScale, DateScale, DateColorScale, OrdinalScale,
    OrdinalColorScale, ColorScale, Scatter, Lines, Figure, Tooltip
)
from ipywidgets import Label

## Get Data

price_data = pd.DataFrame(np.cumsum(np.random.randn(150, 2).dot([[1.0, -0.8], [-0.8, 1.0]]), axis=0) + 100,
                          columns=['Security 1', 'Security 2'], index=pd.date_range(start='01-01-2007', periods=150))
size = 100
np.random.seed(0)
x_data = range(size)
y_data = np.cumsum(np.random.randn(size) * 100.0)
ord_keys = np.array(['A', 'B', 'C', 'D', 'E', 'F'])
ordinal_data = np.random.randint(5, size=size)

symbols = ['Security 1', 'Security 2']

dates_all = price_data.index.values
dates_all_t = dates_all[1:]
sec1_levels = np.array(price_data[symbols[0]].values.flatten())
log_sec1 = np.log(sec1_levels)
sec1_returns = log_sec1[1:] - log_sec1[:-1]

sec2_levels = np.array(price_data[symbols[1]].values.flatten())

sc_x = LinearScale()
sc_y = LinearScale()
sc_y2 = LinearScale()

sc_size = LinearScale()
sc_opacity = LinearScale()


scatter2 = Scatter(x=sec2_levels[1:], y=sec1_levels, size=sec1_returns,
                   scales={'x': sc_x, 'y': sc_y, 'size': sc_size, 'opacity': sc_opacity}, 
                   default_size=128, colors=['orangered'], stroke='black')

ax_y = Axis(label='Security 1', scale=sc_y, orientation='vertical', side='left')
ax_x = Axis(label='Security 2', scale=sc_x)

Figure(axes=[ax_x, ax_y], marks=[scatter2])

```
